Packs/LogRhythmRest/Integrations/LogRhythmRest/LogRhythmRest.py

```diff
@@ -583,6 +584,1316 @@ def get_alarm_events(data_args):
     return_outputs(readable_output=human_readable, outputs=outputs, raw_response=res)
 
 
+def fetch_incidents():
+    headers = dict(HEADERS)
+
+    last_run = demisto.getLastRun()
+
+    #Check if first run. If not, continue running from the last case dateCreated field.
+    if last_run and 'start_time' in last_run:
+        start_time = last_run.get('start_time')
+        headers['createdAfter'] = start_time
+        #print(start_time)
+    else:
+        headers['createdBefore'] = datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    #Get list of cases
+    cases = http_request('GET','lr-case-api/cases?entityNumber='+str(ENTITY_ID),headers=headers)
+
+    ##Set Last Run to the last case dateCreated field
+    if cases != []:
+        demisto.setLastRun({
+            'start_time': cases[len(cases)-1]['dateCreated']
+        })
+
+
+    ##Generate incidents
+    incidents = []
+    for case in cases:
+        incident = {
+            'name': 'Case #' + str(case['number']) + ' '+ str(case['name']),
+            'occurred': str(case['dateCreated']),
+            'rawJSON': json.dumps(case)
+        }
+        incidents.append(incident)
+
+    demisto.incidents(incidents)
+
+
+def lr_case_query(data_args):
+    case_id = data_args.get('case-id')
+    case = http_request('GET','lr-case-api/cases/'+case_id+'/evidence')
+    result = CommandResults(
+        outputs_prefix='Evidence',
+        outputs = case
+        )
+    return_results(result)
+
+
+
+def lr_search_data(data_args):
+    number_of_date = data_args.get('number-of-date')
+    source_entity = data_args.get('source-entity')
+    source_type = data_args.get('source-type')
+    host_name = data_args.get('host-name')
+    username = data_args.get('username')
+    subject = data_args.get('subject')
+    sender = data_args.get('sender')
+    recipient = data_args.get('recipient')
+    Hash = data_args.get('hash')
+    url = data_args.get('URL')
+    process_name = data_args.get('process-name')
+    Object = data_args.get('object')
+
+
+    #Mapping type and name fields
+
+    source_type_map = {
```

Comment:
this should be defined out of the method in the globals section after `ALARM_STATUS`